### DISCUSSION TITLE
Tweak `zkEVM-` prefix from library names (CMake)

### DISCUSSION THREAD

**Discussion by aleasims** [NilFoundation/zkEVM-framework#77] [Ideas]
Now we agreed to name libraries like `zkEVM<LIBNAME>`. This is actually slightly inconvenient, because:

- zkEVM-framework sources are not designed to be included as a dependency into other project. Possible user would rather build and install it. With that in mind, having library names like `DataTypes` and `MPT` is fine for us, because we will unlikely face name collisions. So, this prefix is not really needed;
- When using installed version of zkEVM-framework, imported with CMake, you will get namespaced names like:

    ```
    zkEVM-framework::MPT
    ```

    And with if the prefix will be there, you would get

    ```
    zkEVM-framework::zkEVMMPT
    ```

    which is a bit awkward;
- `zkEVM` is not a precise prefix, it probably should be `zkEVM-framework` instead.

What we can do:

- Get rid of this prefix completely. So instead of target `zkEVMDataTypes` we will have `DataTypes` as a build target and `zkEVM-framework::DataTypes` as exported target;

or

- Switch to `zkEVMFramework` prefix. So instead of `zkEVMDataTypes` we will have `zkEVMFrameworkDataTypes`.

**Note:** target exporting is not yet implemented. See #27

**Comment by aleasims** [comment]
@makxenov @akokoshn you are welcome to this discussion :)

**Reply by akokoshn** [reply]
For me best naming is `zkEVM-framework::DataTypes`
Nobody uses Framework as external dependency, and most probably will not, so enabling export target just nice to have.
Renaming libraries should be separate task in my understanding

**Reply by aleasims** [reply]
You mean that `zkEVM-framework::DataTypes` should be the name of the target?

```cmake
add_library(zkEVM-framework::DataTypes SHARED)
```

Like this?
